kerastuner/engine/metrics_tracking.py

```diff
@@ -22,136 +22,220 @@
 from tensorflow import keras
 
 
-MetricObservation = collections.namedtuple(
-    'MetricObservation',
-    'value step')
+class Direction(object):
+    def __init__(self, direction='min'):
+        if direction not in {'min', 'max'}:
+            raise ValueError(
+                '`direction` should be one of '
+                '{"min", "max"}, but got: %s' % (direction,))
+        self.direction = direction
+
+    def get_best(self, values):
+        if not values:
+            return None
+        if self.direction == 'min':
+            return np.nanmin(values)
+        return np.nanmax(values)
+
+    def get_config(self):
+        return {'direction': self.direction}
+
+    @classmethod
+    def from_config(cls, config):
+        return cls(config['direction'])
+
+
+class MetricObservation(object):
+
+    def __init__(self, value, step):
+        self.value = value
+        self.step = step
+
+    def append(self, value):
+        if not isinstance(value, list):
+            value = [value]
+        if not isinstance(self.value, list):
+            self.value = [self.value]
+        self.value += value
+
+    def mean(self):
+        if not isinstance(self.value, list):
+            return self.value
+        return np.mean(self.value)
+
+    def get_config(self):
+        return {'value': self.value,
+                'step': self.step}
+
+    @classmethod
+    def from_config(cls, config):
+        return cls(**config)
+
+    def __eq__(self, other):
+        if not isinstance(other, MetricObservation):
+            return False
+        return (other.value == self.value and
+                other.step == self.step)
+
+
+class MetricTracker(object):
```

Comment:
`MetricTracker` and `MetricsTracker` are too close as names, prefer another name like `SingleMetricTracker`

Reply:
Done, changed to `MetricHistory` and `MetricTracker`, wdyt?